nifi-assembly/src/main/assembly/common.xml

```diff
@@ -25,6 +25,8 @@
             <useTransitiveFiltering>true</useTransitiveFiltering>
             <includes>
                 <include>nifi-bootstrap-utils</include>
+                <include>nifi-properties</include>
+                <include>nifi-property-utils</include>
```

Comment:
By including nifi-properties and nifi-property-utils here at this level aren't we making nifi-properties available/accessible in many places?  Including beyond the scope we want?  If it was restricted why are we sure we can expand it?

Reply:
The `DiagnosticContext` class references `BootstrapProperties` and `NiFiBootstrapUtils` for loading properties, which come from these dependencies. The current class design is problematic in that incorporates both property loading and business logic. The implementation could also be refactored to avoid these dependencies.